ingress/controllers/gce/firewalls/firewalls.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package firewalls
+
+import (
+	"github.com/golang/glog"
+
+	compute "google.golang.org/api/compute/v1"
+	"k8s.io/contrib/ingress/controllers/gce/utils"
+	netset "k8s.io/kubernetes/pkg/util/net/sets"
+)
+
+// Src range from which the GCE L7 performs health checks.
+const l7SrcRange = "130.211.0.0/22"
+
+// FirewallRules manages firewall rules.
+type FirewallRules struct {
```

Comment:
It doesn't need to be, that's just a detail. We could well create a rule per nodeport through the Sync function since it has all the information required. It's a "pool" in that it manages a group of cloud resources in a generic way (syncs periodically, shutdown etc). A pool can contain 1 item, but the other way around doesn't hold. In an ideal world all of : backends, instance groups, firewalls etc would satisfy a pool interface, but today that's not really possible unless I use `interface{}` everywhere which is gross, so I decided to just convey the intent through naming and similar methods.